From e7618c3e1c5d3073a8db856fe0929981c3f703da Mon Sep 17 00:00:00 2001
From: "Thulasimani,Sivakumar" <sivakumar.thulasimani@intel.com>
Date: Mon, 21 Dec 2015 13:58:24 +0530
Subject: [PATCH 03/19] FOR_UPSTREAM [VPG]: drm/i915: Avoid lane optimization
 for External panels

Lane count optimization is not required for external panels
also it might be affecting normal working in case of MST
panels, where it is ideal to use as many lanes
available as possible to support any future panel
plugged in through that chain. This patch fixes this
by not optimizing lanes for external panels alone.

Change-Id: I12d34cdfb50e5099ac9067264cd039e6f6d678c0
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-30099
Signed-off-by: Sivakumar Thulasimani <sivakumar.thulasimani@intel.com>
Reviewed-on: https://android.intel.com:443/498178
---
 drivers/gpu/drm/i915/intel_dp.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/gpu/drm/i915/intel_dp.c b/drivers/gpu/drm/i915/intel_dp.c
index aa4bf5a..a56e11d 100644
--- a/drivers/gpu/drm/i915/intel_dp.c
+++ b/drivers/gpu/drm/i915/intel_dp.c
@@ -1074,8 +1074,8 @@ intel_dp_compute_config(struct intel_encoder *encoder,
 			DRM_DEBUG_KMS("using min %02x link bw per VBT\n",
 				      bws[min_clock]);
 		}
-	}
-
+	} else
+		min_lane_count = max_lane_count;
 	/*
 	 * Walk through all bpp values. Luckily they're all nicely spaced with 2
 	 * bpc in between.
-- 
1.9.1

